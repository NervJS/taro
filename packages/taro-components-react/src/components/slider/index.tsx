import './loader'

export { Slider as default } from '@tarojs/components/lib/react'
