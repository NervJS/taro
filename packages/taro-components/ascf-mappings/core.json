{
  "Button": {
    "size": {
      "ascfProp": "dimension",
      "valueMap": {
        "mini": "XS",
        "small": "S",
        "default": "M",
        "large": "L"
      },
      "required": false,
      "default": "M",
      "docLink": "https://ascf.docs/components/button#dimension"
    },
    "loading": {
      "ascfProp": "isLoading",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/button#isloading"
    }
  },
  "Input": {
    "password": {
      "ascfProp": "secretInput",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/input#secretinput"
    }
  }
}
