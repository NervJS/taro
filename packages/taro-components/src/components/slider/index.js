import 'weui'
import Nerv from 'nervjs'
import { isNumber, isBoolean, isString, isFunction } from '../../utils/parse-type'
import classNames from 'classnames'

/**
 * props 类型检测
 *
 * @param {Object} props
 */
function parseType (props) {
  const {
    min,
    max,
    step,
    disabled,
    value,
    backgroundColor,
    activeColor,
    blockSize,
    blockColor,
    showValue,
    onChange,
    onChanging
  } = props

  // 抛出错误信息
  const throwErrorMsg = type => {
    throw new TypeError(type)
  }

  if (min) isNumber(min) ? '' : throwErrorMsg('min')
  if (max) isNumber(max) ? '' : throwErrorMsg('max')
  if (step) isNumber(step) ? '' : throwErrorMsg('step')
  if (value) isNumber(value) ? '' : throwErrorMsg('value')
  if (blockSize) isNumber(blockSize) ? '' : throwErrorMsg('blockSize')

  if (disabled) isBoolean(disabled) ? '' : throwErrorMsg('disabled')
  if (showValue) isBoolean(showValue) ? '' : throwErrorMsg('showValue')

  if (backgroundColor) {
    isString(backgroundColor) ? '' : throwErrorMsg('backgroundColor')
  }

  if (activeColor) isString(activeColor) ? '' : throwErrorMsg('activeColor')
  if (blockColor) isString(blockColor) ? '' : throwErrorMsg('blockColor')

  if (onChange) isFunction(onChange) ? '' : throwErrorMsg('onChange')
  if (onChanging) isFunction(onChanging) ? '' : throwErrorMsg('onChanging')
}

class Slider extends Nerv.Component {
  constructor () {
    super(...arguments)

    parseType(this.props)

    this.sliderInsRef = ''

    if (this.props.value) {
      if (this.props.value > this.props.max) {
        this.props.value = this.props.max
      }
    }

    this.state = {
      value: this.props.value,
      controlled: typeof this.props.value !== 'undefined',
      totalWidth: 0,
      touching: false,
      ogX: 0,
      touchID: false,
      percent: this.props.value
        ? parseInt(this.props.value / (this.props.max - this.props.min) * 100)
        : 0
    }
    this.handleTouchStart = this.handleTouchStart.bind(this)
    this.handleTouchMove = this.handleTouchMove.bind(this)
    this.handleTouchEnd = this.handleTouchEnd.bind(this)
  }

  componentDidMount () {
    if (this.state.value === 0) {
      this.setState({
        value: this.countValue()
      })
    }
  }

  componentWillReceiveProps (nextProps) {
    if (this.state.controlled) {
      if (
        nextProps.value <= this.props.max &&
        nextProps.value >= this.props.min
      ) {
        let percent = parseInt(
          nextProps.value / (this.props.max - this.props.min) * 100
        )
        this.setState({ value: nextProps.value, percent })
      }
    }
  }

  countValue (percent = this.state.percent, min = this.props.min, max = this.props.max, step = this.props.step) {
    let value = 0

    if (percent === 100) {
      value = max
    } else if (percent === 0) {
      value = min
    } else {
      value = Math.round(percent * max / 100 / step)
      value = Math.round(value * step)
    }
    return value
  }

  handleTouchStart (e) {
    if (this.state.touching || this.props.disabled) return
    let barDOM = Nerv.findDOMNode(this.sliderInsRef)
    this.setState({
      touching: true,
      touchId: e.targetTouches[0].identifier,
      totalWidth: barDOM.clientWidth,
      ogX: e.targetTouches[0].pageX,
      ogPercent: this.state.percent
    })
  }

  handleTouchMove (e) {
    let { onChanging } = this.props
    if (!this.state.touching || this.props.disabled) return
    if (e.targetTouches[0].identifier !== this.state.touchId) return

    // 阻止默认事件
    e.preventDefault()

    const pageX = e.targetTouches[0].pageX
    const diffX = pageX - this.state.ogX

    let percent =
      parseInt(diffX / this.state.totalWidth * 100) + this.state.ogPercent
    percent = percent < 0 ? 0 : percent > 100 ? 100 : percent

    const value = this.countValue(percent)

    if (value === this.state.value) return

    this.setState(
      {
        percent,
        value
      },
      () => {
        Object.defineProperty(e, 'detail', {
          enumerable: true,
          value: {
            detail: e.detail,
            value: this.state.value
          }
        })
        if (onChanging) onChanging(e)
      }
    )
  }
  handleTouchEnd (e) {
    if (!this.state.touching || this.props.disabled) {
      return
    }

    let { onChange } = this.props

    this.setState({
      touching: false,
      ogX: 0,
      touchId: false,
      ogPercent: 0
    }, () => {
      Object.defineProperty(e, 'detail', {
        enumerable: true,
        value: {
          detail: e.detail,
          value: this.state.value
        }
      })
      if (onChange) onChange(e)
    })
  }

  render () {
    const {
      name = '',
      className,
      showValue,
      backgroundColor,
      activeColor,
      blockColor,
      blockSize,
      ...restProps
    } = this.props
    let _blockSize = blockSize
    let cls = classNames('weui-slider-box', className)

    let innerStyles = {
      backgroundColor: backgroundColor
    }

    const percent = this.state.percent > 100 ? 100 : this.state.percent
    let trackStyles = {
      width: `${percent}%`,
      backgroundColor: activeColor
    }

    if (_blockSize < 12) {
      _blockSize = 28
    }
    if (_blockSize > 28) {
      _blockSize = 28
    }

    let handlerStyles = {
      left: `${percent}%`,
      width: `${_blockSize}px`,
      height: `${_blockSize}px`,
      backgroundColor: blockColor,
      marginTop: `-${Math.floor(_blockSize / 2)}px`,
      marginLeft: `-${Math.floor(_blockSize / 2)}px`
    }
    return (
      <div className={cls} {...restProps} >
        <div className='weui-slider'>
          <div className='weui-slider__inner' style={innerStyles} ref={c => (this.sliderInsRef = c)}>
            <div style={trackStyles} className='weui-slider__track' />
            <div
              style={handlerStyles}
              onTouchStart={this.handleTouchStart}
              onTouchMove={this.handleTouchMove}
              onTouchEnd={this.handleTouchEnd}
              className='weui-slider__handler'
            />
            <input type='hidden' name={name} value={this.state.value} />
          </div>
        </div>
        {showValue ? (
          <div className='weui-slider-box__value'>{this.state.value}</div>
        ) : (
          false
        )}
      </div>
    )
  }
}

Slider.defaultProps = {
  max: 100,
  min: 0,
  step: 1,
  showValue: false,
  disabled: false,
  value: 0
}
export default Slider
