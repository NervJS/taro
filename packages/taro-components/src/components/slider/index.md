slider

## API

|     | 属性            | 类型        | 默认值  | 说明                                          |
| --- | --------------- | ----------- | ------- | --------------------------------------------- |
| √   | min             | Number      | 0       | 最小值                                        |
| √   | max             | Number      | 100     | 最大值                                        |
| √   | step            | Number      | 1       | 步长，取值必须大于 0，并且可被(max - min)整除 |
| √   | disabled        | Boolean     | false   | 是否禁用                                      |
| √   | value           | Number      | 0       | 当前取值                                      |
|     | color           | Color       | #e9e9e9 | 背景条的颜色（请使用 backgroundColor）        |
|     | selected-color  | Color       | #1aad19 | 已选择的颜色（请使用 activeColor）            |
| √   | activeColor     | Color       | #1aad19 | 已选择的颜色                                  |
| √   | backgroundColor | Color       | #e9e9e9 | 背景条的颜色                                  |
| √   | block-size      | Number      | 28      | 滑块的大小，取值范围为 12 - 28                |
| √   | block-color     | Color       | #ffffff | 滑块的颜色                                    |
| √   | show-value      | Boolean     | false   | 是否显示当前 value                            |
| √   | bindchange      | EventHandle |         | 完成一次拖动后触发的事件                      |
| √   | bindchanging    | EventHandle |         | 拖动过程中触发的事件                          |
