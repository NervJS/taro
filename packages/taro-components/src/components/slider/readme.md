# taro-slider-core

## API

|     | 属性            | 类型        | 默认值  | 说明                                          |
| --- | --------------- | ----------- | ------- | --------------------------------------------- |
| √   | min             | Number      | 0       | 最小值                                        |
| √   | max             | Number      | 100     | 最大值                                        |
| √   | step            | Number      | 1       | 步长，取值必须大于 0，并且可被(max - min)整除 |
| √   | disabled        | Boolean     | false   | 是否禁用                                      |
| √   | value           | Number      | 0       | 当前取值                                      |
|     | color           | Color       | #e9e9e9 | 背景条的颜色（请使用 backgroundColor）        |
|     | selected-color  | Color       | #1aad19 | 已选择的颜色（请使用 activeColor）            |
| √   | activeColor     | Color       | #1aad19 | 已选择的颜色                                  |
| √   | backgroundColor | Color       | #e9e9e9 | 背景条的颜色                                  |
| √   | block-size      | Number      | 28      | 滑块的大小，取值范围为 12 - 28                |
| √   | block-color     | Color       | #ffffff | 滑块的颜色                                    |
| √   | show-value      | Boolean     | false   | 是否显示当前 value                            |
| √   | bindchange      | EventHandle |         | 完成一次拖动后触发的事件                      |
| √   | bindchanging    | EventHandle |         | 拖动过程中触发的事件                          |

<!-- Auto Generated Below -->


## Properties

| Property          | Attribute          | Description | Type      | Default     |
| ----------------- | ------------------ | ----------- | --------- | ----------- |
| `activeColor`     | `active-color`     |             | `string`  | `'#1aad19'` |
| `backgroundColor` | `background-color` |             | `string`  | `'#e9e9e9'` |
| `blockColor`      | `block-color`      |             | `string`  | `'#ffffff'` |
| `blockSize`       | `block-size`       |             | `number`  | `28`        |
| `disabled`        | `disabled`         |             | `boolean` | `false`     |
| `max`             | `max`              |             | `number`  | `100`       |
| `min`             | `min`              |             | `number`  | `0`         |
| `name`            | `name`             |             | `string`  | `''`        |
| `showValue`       | `show-value`       |             | `boolean` | `false`     |
| `step`            | `step`             |             | `number`  | `1`         |
| `value`           | `value`            |             | `number`  | `0`         |


## Events

| Event      | Description | Type               |
| ---------- | ----------- | ------------------ |
| `change`   |             | `CustomEvent<any>` |
| `changing` |             | `CustomEvent<any>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
