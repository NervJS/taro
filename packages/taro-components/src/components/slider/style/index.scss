@import '../../../styles/widget/weui-slider';
