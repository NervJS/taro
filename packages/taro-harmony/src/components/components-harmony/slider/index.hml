<div>
  <slider
    id="{{id}}"
    style="color: {{color}};selected-color: {{selectedColor}};block-color: {{sliderBlockColor}}"
    min="{{min}}"
    max="{{max}}"
    step="{{step}}"
    disabled="{{disabled}}"
    value="{{value}}"
    showsteps="{{showsteps}}"
    showtips="{{showtips}}"
    onchange="onChange"
  ></slider>
  <text if="{{showValue}}" style="width: 60px; font-size: 20px; color: gray">{{currentValue}}</text>
</div>

