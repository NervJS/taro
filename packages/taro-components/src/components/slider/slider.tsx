// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { Component, h, ComponentInterface, Prop, Event, EventEmitter, State, Watch, Host, Element } from '@stencil/core'

@Component({
  tag: 'taro-slider-core'
})
export class Slider implements ComponentInterface {
  private sliderInsRef: HTMLDivElement
  private handler: HTMLDivElement

  @Element() el: HTMLElement

  @Prop() min = 0
  @Prop() max = 100
  @Prop() step = 1
  @Prop() disabled = false
  @Prop({ mutable: true }) value: number | null = 0
  @Prop() activeColor = '#1aad19'
  @Prop() backgroundColor = '#e9e9e9'
  @Prop() blockSize = 28
  @Prop() blockColor = '#ffffff'
  @Prop() showValue = false
  @Prop() name = ''

  @State() val: number
  @State() totalWidth = 0
  @State() touching = false
  @State() ogX = 0
  @State() touchId: number | null = null
  @State() percent = 0
  @State() ogPercent: number

  @Watch('value')
  function (newVal) {
    const { max, min } = this
    if (newVal !== null && newVal !== this.val) {
      const val = Math.max(min, Math.min(newVal, max))
      this.updateByStep(val)
    }
  }

  componentDidLoad () {
    Object.defineProperty(this.el, 'value', {
      get: () => this.val,
      set: value => (this.value = value),
      configurable: true
    })

    // 在自动化测试时，如果通过 JSX 绑定 touch 事件，
    // 模拟的 touch 事件只会在浏览器的 device mode 下触发，Karma 跑的测试就会跪。
    // 因此改为 didLoad 后 addEventListener 的形式。
    this.handler.addEventListener('touchstart', this.handleTouchStart)
    this.handler.addEventListener('touchmove', this.handleTouchMove)
    this.handler.addEventListener('touchend', this.handleTouchEnd)
  }

  componentDidUpdate () {
    this.value = null
  }

  @Event({
    eventName: 'change'
  })
  onChange: EventEmitter

  @Event({
    eventName: 'changing'
  })
  onChanging: EventEmitter

  componentWillLoad () {
    const { value, max, min } = this
    if (value === null) return

    const val = Math.max(min, Math.min(value, max))
    this.updateByStep(val)
  }

  handleTouchStart = (e: TouchEvent) => {
    if (this.touching || this.disabled) return

    this.touching = true
    this.touchId = e.targetTouches[0].identifier
    this.totalWidth = this.sliderInsRef.clientWidth
    this.ogX = e.targetTouches[0].pageX
    this.ogPercent = this.percent
  }

  handleTouchMove = (e: TouchEvent) => {
    const {
      disabled,
      touching,
      touchId,
      totalWidth,
      max,
      min,
      ogX,
      ogPercent
    } = this
    if (!touching || disabled) return
    if (e.targetTouches[0].identifier !== touchId) return

    // 阻止默认事件
    e.preventDefault()

    const pageX = e.targetTouches[0].pageX
    const diffX = pageX - ogX

    let percent = diffX / totalWidth * 100 + ogPercent
    percent = Math.max(0, Math.min(percent, 100))
    const val = min + percent * 0.01 * (max - min)

    this.updateByStep(val)

    this.onChanging.emit({
      detail: e.detail,
      value: this.val
    })
  }

  handleTouchEnd = (e: TouchEvent) => {
    const { disabled, touching } = this
    if (!touching || disabled) return

    if (this.percent !== this.ogPercent) {
      this.onChange.emit({
        detail: e.detail,
        value: this.val
      })
    }

    this.touching = false
    this.touchId = null
    this.ogX = 0
    this.ogPercent = 0
  }

  // 根据步长 step 修改 value
  updateByStep (value: number) {
    const { max, min, step } = this
    const steps = Math.floor((max - min) / step)
    for (let i = 0; i <= steps; i++) {
      const current = min + step * i
      const next = i === steps ? null : min + step * (i + 1)

      if (value === current) break

      if (!next && value > current) {
        // step 不能被 max - min 整除
        value = current
      }

      if (next && value > current && value < next) {
        if (value - current < step / 2) {
          value = current
        } else {
          value = next
        }
        break
      }
    }

    const percent = (value - min) / (max - min) * 100

    this.val = value
    this.percent = percent
  }

  render () {
    const {
      showValue,
      backgroundColor,
      activeColor,
      blockColor,
      name,
      percent,
      val
    } = this
    let blockSize = this.blockSize

    const innerStyles = { backgroundColor }
    const percentage = percent > 100 ? 100 : percent
    const trackStyles = {
      width: `${percentage}%`,
      backgroundColor: activeColor
    }

    if (blockSize < 12) {
      blockSize = 12
    }
    if (blockSize > 28) {
      blockSize = 28
    }

    const handlerStyles = {
      left: `${percentage}%`,
      width: `${blockSize}px`,
      height: `${blockSize}px`,
      backgroundColor: blockColor,
      marginTop: `-${Math.floor(blockSize / 2)}px`,
      marginLeft: `-${Math.floor(blockSize / 2)}px`
    }
    return (
      <Host class='weui-slider-box'>
        <div class='weui-slider'>
          <div class='weui-slider__inner' style={innerStyles} ref={c => (this.sliderInsRef = c!)}>
            <div style={trackStyles} class='weui-slider__track' />
            <div
              class='weui-slider__handler'
              ref={dom => {
                if (dom) this.handler = dom
              }}
              style={handlerStyles}
            />
            <input type='hidden' name={name} value={val} />
          </div>
        </div>
        {showValue && <div class='weui-slider-box__value'>{val}</div>}
      </Host>
    )
  }
}
