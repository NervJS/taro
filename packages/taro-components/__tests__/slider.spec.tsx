import { h } from '@stencil/core'
import { AnyHTMLElement } from '@stencil/core/internal'
import { newSpecPage, SpecPage } from '@stencil/core/testing'

import { Slider } from '../src/components/slider/slider'

describe('Slider', () => {
  let page: SpecPage

  it('should be max', async () => {
    const max = 200
    const current = 300
    page = await newSpecPage({
      components: [Slider],
      template: () => (<taro-slider-core
        max={max}
        value={current}
      />),
    })
    await page.waitForChanges()

    expect(page.root?.value).toEqual(max)

    expect(page.root).toMatchSnapshot()
  })

  it('should be min', async () => {
    const min = 50
    const current = 0
    page = await newSpecPage({
      components: [Slider],
      template: () => (<taro-slider-core
        min={min}
        value={current}
      />),
    })

    expect(page.root?.value).toEqual(min)

    expect(page.root).toMatchSnapshot()
  })

  it('steps', async () => {
    page = await newSpecPage({
      components: [Slider],
      template: () => (<taro-slider-core min={50} max={200} step={2} value={79} />),
    })
    const track = page.root?.querySelector<AnyHTMLElement>('.weui-slider__track')
    const handler = page.root?.querySelector<AnyHTMLElement>('.weui-slider__handler')

    expect(page.root?.value).toEqual(80)
    expect(track?.style.width).toEqual('20%')
    expect(handler?.style.left).toEqual('20%')

    expect(page.root).toMatchSnapshot()
  })

  it('events', async () => {
    const onChange = jest.fn()
    const onChanging = jest.fn()
    page = await newSpecPage({
      components: [Slider],
      template: () => (<taro-slider-core onChange={onChange} onChanging={onChanging} />),
    })

    const inner = page.root?.querySelector<AnyHTMLElement>('.weui-slider__inner')
    const handler = page.root?.querySelector<AnyHTMLElement>('.weui-slider__handler')
    const innerWidth = inner?.getBoundingClientRect().width || 0
    const rect = handler?.getBoundingClientRect() || { left: 0, right: 0 }
    const centerPoint = (rect.left + rect.right) / 2

    await page.waitForChanges()
    handler?.dispatchEvent(new Event('touchstart', {
      // @ts-ignore
      targetTouches: [{
        identifier: 0,
        target: handler,
        pageX: centerPoint
      }]
    }))

    handler?.dispatchEvent(new Event('touchmove', {
      // @ts-ignore
      targetTouches: [{
        identifier: 0,
        target: handler,
        pageX: centerPoint + innerWidth / 2
      }]
    }))

    handler?.dispatchEvent(new Event('touchend'))

    const step = 0 // 50
    expect(onChanging).toBeCalledTimes(1)
    expect(onChanging.mock.instances[0].value).toEqual(step)
    expect(page.root?.value).toEqual(step)

    expect(page.root).toMatchSnapshot()
  })
})
