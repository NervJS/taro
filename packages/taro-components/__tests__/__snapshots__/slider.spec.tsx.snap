// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider events 1`] = `
<taro-slider-core class="weui-slider-box" value="0">
  <div class="weui-slider">
    <div class="weui-slider__inner" style="background-color: #e9e9e9;">
      <div class="weui-slider__track" style="width: 0%; background-color: #1aad19;"></div>
      <div class="weui-slider__handler" style="left: 0%; width: 28px; height: 28px; background-color: #ffffff; margin-top: -14px; margin-left: -14px;"></div>
      <input type="hidden" value="0">
    </div>
  </div>
</taro-slider-core>
`;

exports[`Slider should be max 1`] = `
<taro-slider-core class="weui-slider-box" value="200">
  <div class="weui-slider">
    <div class="weui-slider__inner" style="background-color: #e9e9e9;">
      <div class="weui-slider__track" style="width: 100%; background-color: #1aad19;"></div>
      <div class="weui-slider__handler" style="left: 100%; width: 28px; height: 28px; background-color: #ffffff; margin-top: -14px; margin-left: -14px;"></div>
      <input type="hidden" value="200">
    </div>
  </div>
</taro-slider-core>
`;

exports[`Slider should be min 1`] = `
<taro-slider-core class="weui-slider-box" value="50">
  <div class="weui-slider">
    <div class="weui-slider__inner" style="background-color: #e9e9e9;">
      <div class="weui-slider__track" style="width: 0%; background-color: #1aad19;"></div>
      <div class="weui-slider__handler" style="left: 0%; width: 28px; height: 28px; background-color: #ffffff; margin-top: -14px; margin-left: -14px;"></div>
      <input type="hidden" value="50">
    </div>
  </div>
</taro-slider-core>
`;

exports[`Slider steps 1`] = `
<taro-slider-core class="weui-slider-box" value="80">
  <div class="weui-slider">
    <div class="weui-slider__inner" style="background-color: #e9e9e9;">
      <div class="weui-slider__track" style="width: 20%; background-color: #1aad19;"></div>
      <div class="weui-slider__handler" style="left: 20%; width: 28px; height: 28px; background-color: #ffffff; margin-top: -14px; margin-left: -14px;"></div>
      <input type="hidden" value="80">
    </div>
  </div>
</taro-slider-core>
`;
